__tests__/app.test.js

```diff
@@ -31,9 +33,377 @@ describe("NCNews-Server Unit Tests", () => {
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an invalid username`, () => {
+
+            const newComment = {
+                username: 545435345,
+                body: "k"
+            };
+
+            return request(app).post('/api/articles/1/comments').send(newComment).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an SQL injection in the JSON object`, () => {
+
+            const newComment = {
+                username: `icellusedkars; DROP TABLE articles;`,
+                body: `blah; DROP TABLE articles;`
+            };
+
+            return request(app).post('/api/articles/1/comments').send(newComment).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an SQL injection in URL parameters`, () => {
+
+            return request(app).post(`/api/articles/1; DROP TABLE articles;/comments`).send(defaultComment).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+        });
+
+
+    });         
+
+    // Task 8 - PATCH /api/articles/:article_id
+    describe(`PATCH /api/articles/:article_id`, () => {
+
+        const defaultVotesObject = { inc_votes : 1 };
+
+        test(`[ 201 ] Responds with the updated article with correctly adjusted vote count, increasing vote count`, () => {
+            const votesObject = { inc_votes : 2 };
+
+            return request(app).patch('/api/articles/1').send(votesObject).expect(201).then((response) => {
+                
+                expect(response.body.article["article_id"]).toEqual(1);
+                expect(response.body.article.votes).toEqual(102);
+            });
+
+        });
+
+        test(`[ 201 ] Responds with the updated article with correctly adjusted vote count, decreasing vote count`, () => {
+            const votesObject = { inc_votes : -5 };
+
+            return request(app).patch('/api/articles/1').send(votesObject).expect(201).then((response) => {
+
+                expect(response.body.article["article_id"]).toEqual(1);
+                expect(response.body.article.votes).toEqual(95);
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed invalid URL parameters`, () => {
+
+            return request(app).patch('/api/articles/fddfggdrtd').send(defaultVotesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an object without the required keys`, () => {
```

Comment:
This is personal preference but I think this could be a 200 but not increase the votes, this would allow you to enable this patch end point to be extended in the future, if we wanted to add the ability to patch the body for example